### PR TITLE
missing product gamma when calc discount_returns; reduce_mean twice f…

### PR DESCRIPTION
**Pull request by witwolf**
…or entropy loss, not equal with mask of LAST step

### CODE REVIEW

File: alf/algorithms/actor_critic_loss.py [hunk 1]
```diff
@@ -121,8 +120,8 @@ def __call__(self, training_info: TrainingInfo, value, final_time_step,
         if self._entropy_regularization is not None:
             entropies = tfa_common.entropy(training_info.action_distribution,
                                            self._action_spec)
-            entropy_loss = -tf.reduce_mean(input_tensor=entropies)
-            loss += self._entropy_regularization * entropy_loss
+            entropy_loss = -self._entropy_regularization * entropies
```
Comment (emailweixu) [inline alf/algorithms/actor_critic_loss.py]: Keep entropy_loss to the original negative entropy: `entropy_loss = - entropies` so that we can see the original entropy in tensorboard. Apply the weight at the next line.
